cmd/scaffold.go

```diff
@@ -682,15 +684,32 @@ func (fnb *FlowNodeBuilder) Module(name string, f func(builder NodeBuilder, node
 	return fnb
 }
 
-// MustNot asserts that the given error must not occur.
-//
-// If the error is nil, returns a nil log event (which acts as a no-op).
-// If the error is not nil, returns a fatal log event containing the error.
-func (fnb *FlowNodeBuilder) MustNot(err error) *zerolog.Event {
+// Errors returns a channel that receives any unrecoverable errors encountered by the builder
+func (fnb *FlowNodeBuilder) Errors() <-chan error {
+	return fnb.errorManager.Errors()
+}
+
+// handle an unrecoverable error. if an error manager is configured, pass error to the manager, otherwise log a panic
+func (fnb *FlowNodeBuilder) ThrowError(err error, msg string) {
```

Comment:
I would suggest keeping the API as simply `ThrowError(err error)`.

Then, instead of `fnb.ThrowError(nil, "message")`, callers can just do `fnb.ThrowError(errors.New("message"))`.

I noticed that the first argument is `nil` for a large percentage of the calls, which suggests that the current API deserves to be modified.

Alternatively, you could split this into two separate methods `Throw(msg string)` and `ThrowError(err error, msg string)`

Reply:
I was hesitant to squash the error and message before checking if we should panic since that would change the behaviour for all the nodes that don't run with the error manager. I guess that's going to happen sooner or later when we implement this for the other node types anyway, so it's probably cleaner to just rip the bandaid off now.